x/gov/legacy_0_33/genesis.go

```diff
@@ -0,0 +1,42 @@
+package gov_0_33
```

Comment:
Why do we have this? Is the genesis state different? I don't think the current directory structure is correct.

Reply:
The genesis state contains the genesis proposals and the format of proposals has been changed in this pr. This will not persist, we can remove the migration files at the release after it has been included(though it is bit PM side decision, I don't have hard opinion)

Reply:
I agree that this is ugly but I think it is worth keeping the old types around for genesis conversion. Can you think of a cleaner way to do it @alexanderbez?

Reply:
I would first figure out the bare minimum that we need (types, functions, methods). Is it just genesis types? Then, I think we should just simply add them to a simple `x/gov/legacy` sub-package (exclude the version).

Reply:
I think we will need to include the version _somewhere either in the filename or package name... how will we manage as there are several versions with the same type names within them right? I don't think we should rely on assuming the version of `legacy/`

Reply:
Agreed. I suggested on today's call that we move it to `legacy/` and inside said package have a `doc.go` or `README.md` outline the what and the way -- this would include the version.

wrt to multiple "legacy" bundles, I really don't hope we envision us keeping all of these. We only need these for import/export of cosmoshub1 => cosmoshub2. Future updates will just replace these contents. Git will keep the history for previous ones if we need to reference them.

Reply:
RE multiple-legacies - yeah I recognize this - it's just going to get funny/potentially-confusing when we have disjointed legacies for example:
 - cosmoshub1 => cosmoshub2 upgrades `x/params`
 - cosmoshub2 => cosmoshub3 upgrades `x/staking` 

in this scenario we should probably remove `x/params/types/legacy` in `cosmoshub3` so only one legacy version exists across the repo at any given release... make sense?

Reply:
yeah since this only really pertains to genesis logic, we can have hub directories/packages (if that is what you mean) 👍

Reply:
Agreed on "transient" migration code - single release must have code migrating only from the last release to the current one, which should be deleted on the next release.